cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java

```diff
@@ -191,8 +191,8 @@ private boolean isValidKafkaTopic(String key) {
 
   private boolean isValidPartitionKey(String key) {
     try {
-      LogPartitionType.valueOf(key);
-    } catch (IllegalArgumentException e) {
+      LogPartitionType.valueOf(key.toUpperCase());
+    } catch (IllegalArgumentException | NullPointerException e) {
```

Comment:
The question is how deep in the code base that we practice this. Do we only do this on master startup check? As there are many places that we call `cConf.get()` (or any `cConf.getXXX` variants, which can throw other runtime exceptions, such as `NumberFormatException`), which the config is not validated as part of Master Conf Check. If the same inconsistency arise, those place would still throw NPE with a `null` message (or similarly cryptic runtime exception).